### DISCUSSION TITLE
New Strawberryshake project won't build

### DISCUSSION THREAD

**Discussion by Xeviva** [ChilliCream/hotchocolate#6085] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Strawberry Shake

### Describe the bug

Following the get started guide (https://chillicream.com/docs/strawberryshake/v13/get-started/console) doesn't work.

I made a fresh solution, followed the guide but I get a build error. The code won't compile.

### Steps to reproduce

Reproduction steps:

1. Installed Visual Studio 2022
2. Followed this: https://chillicream.com/docs/strawberryshake/v13/get-started/console to create a new solution and console app project, using chilli cream's Conference gql server (targeting .Net 6.0)
5. Attempted to build solution as per step 5

Expected:
Code compiles

Actual:
Build error:
`error MSB4057: The target "GenerateGraphQLCode" does not exist in the project.`

### Relevant log output

```shell
Build started...
1>------ Build started: Project: strawberryTest, Configuration: Debug Any CPU ------
Restored D:\CodeRepos\consoleapp\strawberryTest\strawberryTest\strawberryTest.csproj (in 26 ms).
1>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\Roslyn\Microsoft.CSharp.Core.targets(57,11): error MSB4057: The target "GenerateGraphQLCode" does not exist in the project.
1>Done building project "strawberryTest.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build started at 6:43 PM and took 01.690 seconds ==========
```

### Additional Context?

none

### Version

13.0.5

**Comment by michaelstaib** [comment]
Can you attach a demo project here showing this issue?

**Comment by michaelstaib** [comment]
From the error message it looks like the StrawberryShake.Server package is missing in your project.

**Comment by Xeviva** [comment]
I'm now running into what looks like an issue with the documents setting in .graphqlrc.json

If I set documents to `"**/*.graphql", `
I get:  EXEC : error GQL: An item with the same key has already been added. Key: JSON

If I set documents to `"./Documents/*.graphql",` and put queries in a Folder called Documents:
Warning	MSB5029	The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "<somePAth>.nuget\packages\strawberryshake.server\13.0.5\build\StrawberryShake.Server.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always defined.
This occurs with .graphqlrc.json
and no client seems to be generated

If I set documents to `"GQLDocs/*.graphql"` and put docs in GQLDocs:
allows the project to build with no errors, but still no client generated, so I can't add the client to the DI, using builder.Services.AddXClient(), with XClient being the name, even with a using for the namespace value set in .graphqlrc.json

can you advise what should be set here as the default isn't working?

`{
  "schema": "schema.graphql",
  "documents": "GQLDocs/*.graphql",
  "extensions": {
    "strawberryShake": {
      "name": "XClient",
      "namespace": "XGQL",
      "url": "https://someserver.com/graphql",
      "records": {
        "inputs": false,
        "entities": false
      },
      "transportProfiles": [
        {
          "default": "Http",
          "subscription": "WebSocket"
        }
      ]
    }
  }
}`

**Comment by michaelstaib** [comment]
join us on slack.chillicream.com and ping me there ....

**Comment by mnj** [comment]
Same here, I got .NET 6, 7 and 8 preview versions installed, the rest of the solution projects are all .NET 8, even if create a class lib with .NET 7, it just fails completely on dotnet build, when following the (sorry, quite lacking) documentation:

```
PS C:\Projects\XXXX\XXXX.YYYY.GithubService\GithubService.GithubGraphQL> dotnet build                                     
MSBuild version 17.8.0-preview-23367-03+0ff2a83e9 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
C:\Program Files\dotnet\sdk\8.0.100-preview.7.23376.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(314,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [C:\Projects\XXXX\XXXX.YYYY.GithubService\Git
hubService.GithubGraphQL\GithubService.GithubGraphQL.csproj]
  Could not execute because the specified command or file was not found.
  Possible reasons for this include:
    * You misspelled a built-in dotnet command.
    * You intended to execute a .NET program, but dotnet-C:\Users\PC\.nuget\packages\strawberryshake.server\13.4.0\build\..\tools\net8\dotnet-graphql.dll does not exist.
    * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.
C:\Users\PC\.nuget\packages\strawberryshake.server\13.4.0\build\StrawberryShake.Server.targets(79,5): error MSB3073: The command "dotnet "C:\Users\PC\.nuget\packages\strawberryshake.server\13.4.0\build\..\tools\net8\dotnet-graphql.dll" generate "C:\Projects\XXXX\XXXX.YYYY.GithubService\Git
hubService.GithubGraphQL" -o "C:\Projects\XXXX\XXXX.YYYY.GithubService\GithubService.GithubGraphQL\obj\Debug\net7.0\berry" -n GithubService.GithubGraphQL -a md5 -t" exited with code 1. [C:\Projects\XXXX\XXXX.YYYY.GithubService\GithubService.GithubGraphQL\GithubService.Gith 
ubGraphQL.csproj]

Build FAILED.

C:\Users\PC\.nuget\packages\strawberryshake.server\13.4.0\build\StrawberryShake.Server.targets(79,5): error MSB3073: The command "dotnet "C:\Users\PC\.nuget\packages\strawberryshake.server\13.4.0\build\..\tools\net8\dotnet-graphql.dll" generate "C:\Projects\XXXX\XXXX.YYYY.GithubService\Git 
hubService.GithubGraphQL" -o "C:\Projects\XXXX\XXXX.YYYY.GithubService\GithubService.GithubGraphQL\obj\Debug\net7.0\berry" -n GithubService.GithubGraphQL -a md5 -t" exited with code 1. [C:\Projects\XXXX\XXXX.YYYY.GithubService\GithubService.GithubGraphQL\GithubService.Gith 
ubGraphQL.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.17
```

**Reply by mnj** [reply]
Ok resolved it by adding a global.json for .net 7, in the project root folder, i really feel like this should be in the getting started pages tbh.

Edit: I take that back, it correctly writes:
 Generate C# Clients started.
  Generate GithubClient started.
  Generate GithubClient completed in 78 ms
  Generate C# Clients completed in 239 ms

But there is no Generated folder, and the assembly contains nothing, so seems like its doing nothing after all.
